### PR TITLE
bluetooth: controller: openisa/RV32M1:  radio Deep Sleep Mode (DSM) 

### PR DESCRIPTION
**Pull request by George-Stefan**
Added the functions to turn on/off the radio from LLL.
Added WAKE_IRQ to start radio after it's out of DSM.
Added the functions to turn on/off the radio.

### CODE REVIEW

**Comment by George-Stefan** [comment]
^^ @alexandru-porosanu-nxp

File: subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/radio/radio.c [hunk 1]
```diff
@@ -103,6 +130,26 @@ static void get_isr_latency(void)
...
 
+static u32_t radio_tmr_start_hlp(u8_t trx, u32_t ticks_start, u32_t remainder);
+static void radio_tmr_hcto_configure_hlp(u32_t hcto);
+static void radio_config_after_wake(void)
+{
+	if (!delayed_radio_start) {
+		delayed_radio_stop = 0;
+		return;
+	}
+	delayed_radio_start = 0;
```
Comment (joerchan) [inline subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/radio/radio.c]: style: blank line after `}`

File: subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/radio/radio.c [hunk 2]
```diff
@@ -351,13 +413,26 @@ void radio_setup(void)
...
 
+	/* Turn radio on to measure ISR latency */
+	if (radio_is_off()) {
+		dsm_ref = 0;
+		radio_wake();
+		while (radio_is_off()) {
+		}
+	} else {
+		dsm_ref = 1;
+	}
```
Comment (joerchan) [inline subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/radio/radio.c]: style: blank line after `}`

File: subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/radio/radio.c [hunk 3]
```diff
@@ -856,6 +931,14 @@ u32_t radio_tmr_start(u8_t trx, u32_t ticks_start, u32_t remainder)
 	}
 	remainder /= 1000000UL;
 
+	if (radio_is_off()) {
+		delayed_radio_start = 1;
+		delayed_trx = trx;
+		delayed_ticks_start = ticks_start;
+		delayed_remainder = remainder;
+		return remainder;
+	}
```
Comment (joerchan) [inline subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/radio/radio.c]: style: blank line after `}`

File: subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/radio/radio.c [hunk 4]
```diff
@@ -1000,3 +1094,96 @@ u32_t radio_ar_has_match(void)
...
+		/* MAN wakeup request enable */
+		RSIM->DSM_CONTROL |= RSIM_DSM_CONTROL_MAN_WAKEUP_REQUEST_EN(1);
+
+		/* Enable DSM, sending a sleep request */
+		GENFSK->DSM_CTRL |= GENFSK_DSM_CTRL_GEN_SLEEP_REQUEST(1);
+
+		/* Enable DSM_TIMER */
+		RSIM->DSM_CONTROL |= RSIM_DSM_CONTROL_DSM_TIMER_EN(1);
+	}
+	return 0;
```
Comment (joerchan) [inline subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/radio/radio.c]: style: blank line after `}`

File: subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/radio/radio.c [hunk 5]
```diff
@@ -902,6 +984,18 @@ void radio_tmr_hcto_configure(u32_t hcto)
...
 }
 
+/* Header completion time out */
+void radio_tmr_hcto_configure(u32_t hcto)
+{
+	if (delayed_radio_start) {
+		delayed_radio_stop = 1;
+		delayed_hcto = hcto;
+		return;
+	}
```
Comment (joerchan) [inline subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/radio/radio.c]: style: blank line after `}`

File: subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/radio/radio.c (hunk 1 above)
Comment (George-Stefan) [reply subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/radio/radio.c]: will fix and rebase

**Review by joerchan — COMMENTED** [review]
Just style comments :)

**Review by cvinayak — APPROVED** [review]
Quick Ack.

**Review by dleach02 — APPROVED** [review]

**Comment by George-Stefan** [comment]
@jhedberg should I rebase?

**Comment by jhedberg** [comment]
> @jhedberg should I rebase?

@George-Stefan AFAIK Shippable does an implicit rebase whenever it runs, so if it passes now then I don't think there's a need.

**Comment by George-Stefan** [comment]
Shippable failures don't seem related to this PR.
I'll rebase anyway to start the tests again

**Review by MaureenHelm — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**